demo/lightning/image-segmentation/model.py

```diff
@@ -58,6 +60,7 @@ def training_step(self, train_batch, batch_idx):
         if self.benchmark:
             # Returning None will break distributed training, so
             # return a scalar of value 1 that represents loss to skip training.
+            time.sleep(self.step_time)
```

Comment:
This approach is probably ok for now and for our benchmarks, but try to measure the actual step time if you're running it this way since it will end up being longer than the configured step time.

The approach for the Tessellations benchmarks is for this to control essentially a "minimum step time" which starts counting from when the step begins (i.e. from the very start of training_step) and then sleeps only on rank 0 if there is still time needed to get to the step time. Sleeping on all ranks can make the actual step time longer since they may not start sleeping at the exact same time. Additionally, I think that the configured step time for those benchmarks may include the data loading time (i.e. the time it takes to get train_batch) but we'd need to double check with Bernard or the team to see if that's really true.

But for getting our quick numbers documented, I think this approach is fine for now.

Reply:
Noted. Two changes to make as discussed in our meeting today:
1. Sleep only if the node's rank is 0
2. Estimate the actual step time from total training time or per-epoch time. Document that instead of/along with of the configured `step_time`